### PR TITLE
Pickers: Fixed infinite loop when focus lost.

### PR DESCRIPTION
**Pull request by joschect**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ npm run change`

#### Description of changes
Pickers: Fixed an issue where a loop would sometimes occur if values were resolved when input had an empty value after an option was selected.

#### Focus areas to test

(optional)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/pickers/BasePicker.tsx [hunk 1]
```diff
@@ -528,6 +525,7 @@ export class BasePicker<T, P extends IBasePickerProps<T>> extends BaseComponent<
       let newItems: T[] = this.state.items.concat([processedItemObject]);
       this._updateSelectedItems(newItems);
     }
+    this.setState({ suggestedDisplayValue: '' });
```
Comment (Lego6245) [inline packages/office-ui-fabric-react/src/components/pickers/BasePicker.tsx]: Is this addition related to the fix, or is this just some housekeeping?

Comment (joschect) [inline packages/office-ui-fabric-react/src/components/pickers/BasePicker.tsx]: Housekeeping. It ensures that when you click out of the picker that and it queues up an item to autoselect that it won't show incorrect text briefly.

File: common/changes/@uifabric/styling/pickers-fixstackoverflow_2017-08-15-22-39.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
+  "changes": [
+    {
+      "comment": "",
+      "packageName": "@uifabric/styling",
+      "type": "none"
```
Comment (dzearing) [inline common/changes/@uifabric/styling/pickers-fixstackoverflow_2017-08-15-22-39.json]: minor suggestion, run `rush change -b upstream/master` to avoid adding the unnecessary change files.

Comment (joschect) [inline common/changes/@uifabric/styling/pickers-fixstackoverflow_2017-08-15-22-39.json]: I'll do that in the future.
